lib/Image/Optimizer.php

```diff
@@ -72,6 +76,12 @@ public function optimizeImage($path)
             unlink($tmpFile['path']);
         }
 
+        foreach ($failedImages as $failedImage) {
+            if (is_file($failedImage)) {
+                unlink($failedImage);
+            }
+        }
+
         if (is_file($workingPath)) {
```

Comment:
Is this not always false? The workingPath seems to be a dir not a file

Reply:
I think `workingPath` is just a misleading variable name, it seems to be a temporary file to work on with the contents from Storage (could be cloud-based): `file_put_contents($workingPath, $storage->read($path));`

Reply:
You are right. Thanks for explaining.